scripts/zkai_native_seq32_attention_mlp_adjacent_label_policy_gate.py

```
@@ -0,0 +1,733 @@
+        "worst_probe_path_opening_saving_vs_champion": 1_232,
+        "best_probe_path_opening_saving_vs_champion": 4_032,
+        "nanozk_reported_d128_block_proof_bytes": NANOZK_REPORTED_D128_BLOCK_PROOF_BYTES,
+        "proof_size_comparable_external_rows": 0,
+    }
+    if summary != expected:
+        raise AdjacentLabelPolicyGateError("summary drift")
+
+
+def validate_variants(variants: list[Any]) -> None:
+    if len(variants) != len(EXPECTED_ROWS):
+        raise AdjacentLabelPolicyGateError("variant inventory drift")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)